.circleci/config.yml

```diff
@@ -1,70 +1,168 @@
+                curl -LO https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubectl
+                chmod +x ./kubectl
+                sudo mv ./kubectl /usr/local/bin/kubectl
+                kubectl version --client
+
+      create-cluster:
+        parameters:
+          kind-config:
+            type: string
+        steps:
+          - run:
+              name: "Create cluster"
+              command: |
+                cd ~/project
+                
+                if [ -e << parameters.kind-config >> ]; then
+                  cp << parameters.kind-config >> kind-config.yaml
+                else
+                  echo "kind: Cluster" >> kind-config.yaml
+                  echo "apiVersion: kind.sigs.k8s.io/v1alpha3" >> kind-config.yaml
+                fi
+
+                kind create cluster --config kind-config.yaml --image quay.io/giantswarm/kind-node:$KUBERNETES_VERSION
+
+      test-setup:
+        parameters:
+          setup-script:
+            type: string
+        steps:
+          - run:
+              name: "Run test setup commands"
+              command: |
+                if [ -e << parameters.setup-script >> ]; then
+                  echo << parameters.setup-script >>
+                  sh << parameters.setup-script >>
+                fi
+
+      go-test:
+        parameters:
+          env:
+            type: string
+          test-dir:
+            type: string
+        steps:
+          - run:
+              name: "Create .env file"
+              command: |
+                echo '<< parameters.env >>' | sed -e 's/^\(.\)/export \1/' > .env
+          - run:
+              # TODO remove this part after switch to Go modules.
+              name: "Move code to $GOPATH"
+              command: |
+                mkdir -p $GOPATH/src/github.com/${CIRCLE_PROJECT_USERNAME} && mv ~/project $GOPATH/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}
+          - run:
+              name: "Run test"
+              command: |
+                cd $GOPATH/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}
+                source .env
+                CGO_ENABLED=0 E2E_KUBECONFIG=~/.kube/config go test -tags k8srequired -timeout 20m -v << parameters.test-dir >>
+
+      export-logs:
+        steps:
+          - run:
+              name: "Export cluster logs"
+              command: |
+                kind export logs ./kind/logs
+              when: always
+          - store_artifacts:
+              path: ./kind/logs
+              destination: logs
+
+    jobs:
+      integration-test:
+        environment:
+          GOPATH: /go
+          KUBERNETES_VERSION: v1.16.3
+        parameters:
+          env:
+            default: ""
+            description: |
+              Environment variables configuration. Must be in following format:
+
+                  VAR_A="VALUE_A"
+                  VAR_B="VALUE_B"
+            type: string
```

Comment:
Do you think it makes more sense to keep those env vars here in the circle configuration or rather specifying it as a path to a file?

Reply:
Yes I think a file is better and matches the other params.